base/bucket_gocb.go

```diff
@@ -1360,6 +1360,53 @@ func (bucket *CouchbaseBucketGoCB) DeleteWithXattr(k string, xattrKey string) er
 
 }
 
+func (bucket *CouchbaseBucketGoCB) GetXattr(k string, xattrKey string, xv interface{}) (casOut uint64, err error) {
+
+	worker := func() (shouldRetry bool, err error, value interface{}) {
+		res, lookupErr := bucket.Bucket.LookupInEx(k, gocb.SubdocDocFlagAccessDeleted).
+			GetEx(xattrKey, gocb.SubdocFlagXattr).Execute()
+
+		switch lookupErr {
+		case nil:
+			res.Content(xattrKey, xv)
+			cas := uint64(res.Cas())
+			return false, err, cas
+		case gocbcore.ErrSubDocBadMulti:
```

Comment:
I didn't think we could get ErrSubDocBadMulti when we're only performing a single GetEx().  Do you have a test that hit this case for GetXattr?

Reply:
One of the tests experiences this. I have added a comment to the test to point out where it is.

Reply:
It would seem that LookupInEx is all that is exposed to us which can only do multi-doc lookups. Apparently they recently looked into the possibility of adding a single lookup function however this resulted in a negligible performance benefit so they opted not to add this.